### PR TITLE
Fix(MyWallet): Review my wallet page and fix UI issues.

### PR DESCRIPTION
**Pull request by abouolia**
- [FIXED] Nothing happens on clicking the "Download QR" button.
- [FIXED] The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type".
- [FIXED] QR popup style.
- [FIXED] Copy and QR text buttons
- [FIXED] "Avaliable" Balance typo.
- [FIXED] "Reservered" Funds typo.
- [ADDED]: "Total balance" to Monero balance.

### CODE REVIEW

File: packages/main/src/services/haveno.ts [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 import fsPromises from "fs/promises";
 import { ipcMain, dialog } from "electron";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (schowdhuri) [inline packages/main/src/services/haveno.ts:20]: Can this be fixed without suppressing eslint?

File: packages/main/src/services/haveno.ts [hunk 2]
```diff
@@ -56,4 +59,30 @@ export function registerHavenoHandlers() {
...
+    if (!file?.filePath) {
+      return 0;
+    }
+    QRCode.toFile(
+      file?.filePath,
+      code,
+      {
+        width: 500,
+      },
+      function (err: Error) {
```
Comment (schowdhuri) [inline packages/main/src/services/haveno.ts]: Arrow function, please

File: packages/main/src/services/haveno.ts [hunk 3]
```diff
@@ -56,4 +59,30 @@ export function registerHavenoHandlers() {
...
+          name: "*",
+        },
+      ],
+      properties: ["createDirectory", "dontAddToRecent"],
+    });
+    if (!file?.filePath) {
+      return 0;
+    }
+    QRCode.toFile(
+      file?.filePath,
```
Comment (schowdhuri) [inline packages/main/src/services/haveno.ts]: Don't need the `?` here

File: packages/main/src/services/haveno.ts [hunk 4]
```diff
@@ -56,4 +59,30 @@ export function registerHavenoHandlers() {
...
+      filters: [
+        {
+          extensions: ["png"],
+          name: "*",
+        },
+      ],
+      properties: ["createDirectory", "dontAddToRecent"],
+    });
+    if (!file?.filePath) {
+      return 0;
```
Comment (schowdhuri) [inline packages/main/src/services/haveno.ts]: `return` should suffice

File: packages/main/src/types/ipc.ts [hunk 5]
```diff
@@ -30,4 +30,6 @@ export enum IpcChannels {
 
   // others
   VerifyAuthToken = "verifyAuthToken",
+
```
Comment (schowdhuri) [inline packages/main/src/types/ipc.ts]: Newline not required

File: packages/renderer/src/components/atoms/Tabs/Tabs.tsx [hunk 6]
```diff
@@ -45,14 +45,14 @@ const useStyles = createStyles((theme, _params, getRef) => {
...
-      "&:first-child": {
+      "&:first-of-type": {
         marginLeft: 0,
       },
       "&:last-child": {
         marginRight: 0,
       },
       [`&.${tabActiveRef}`]: {
-        borderBottomColor: theme.primaryColor,
+        borderBottomColor: theme.colors.blue[6],
```
Comment (schowdhuri) [inline packages/renderer/src/components/atoms/Tabs/Tabs.tsx:55]: Nice catch :)

File: packages/renderer/src/components/atoms/Tabs/Tabs.tsx [hunk 7]
```diff
@@ -45,14 +45,14 @@ const useStyles = createStyles((theme, _params, getRef) => {
       paddingRight: 0,
       textTransform: "uppercase",
 
-      "&:first-child": {
+      "&:first-of-type": {
```
Comment (schowdhuri) [inline packages/renderer/src/components/atoms/Tabs/Tabs.tsx:48]: Nice, thanks!

File: packages/main/src/services/haveno.ts [hunk 8]
```diff
@@ -56,4 +59,30 @@ export function registerHavenoHandlers() {
     const zipFile = files.filePaths[0];
     return new Uint8Array(await fsPromises.readFile(zipFile));
   });
+
+  ipcMain.handle(IpcChannels.DownloadQRCode, async (_, code: string) => {
+    const file = await dialog.showSaveDialog({
```
Comment (schowdhuri) [inline packages/main/src/services/haveno.ts]: Good one!

File: packages/renderer/src/components/atoms/Typography/Anchor.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { Anchor as MAnchor, createStyles } from "@mantine/core";
+import type { AnchorProps as MAnchorProps } from "@mantine/core";
+
+export function Anchor(props: MAnchorProps<"a">) {
```
Comment (schowdhuri) [inline packages/renderer/src/components/atoms/Typography/Anchor.tsx:4]: Good one!

File: packages/renderer/src/components/molecules/AddressCard/AddressCard.tsx [hunk 10]
```diff
@@ -17,20 +17,17 @@
...
-  Group,
-  SimpleGrid,
-  Skeleton,
-} from "@mantine/core";
+import { Box, createStyles, Group, SimpleGrid, Skeleton } from "@mantine/core";
 import { DetailItem } from "@atoms/DetailItem";
 import { Button } from "@atoms/Buttons";
-import { LangKeys } from "@constants/lang";
+import { Anchor } from "@atoms/Typography";
 import { DetailItemCard } from "@atoms/DetailItemCard/DetailItemCard";
```
Comment (schowdhuri) [inline packages/renderer/src/components/molecules/AddressCard/AddressCard.tsx]: `import { DetailItemCard } from @atoms/DetailItemCard`

**Review by schowdhuri — CHANGES_REQUESTED** [review]
Thanks! The UI looks really good now! Minor code style comments added.

File: packages/main/src/services/haveno.ts (hunk 1 above)
Comment (abouolia) [reply packages/main/src/services/haveno.ts:20]: I'm sorry man, I was planning to write to you and explain why i did that, :(, my bad, `qrcode` has two different libraries one for `browser` and another for `server` when i try to import the package in `main` the project try to import it as `browser` package not as `server` package, tried to search about the issue but all solutions say just change your webpack env configuration to `node` (`target: 'node'`), didn't want to modify anything in webpack config. in that PR, so let it that issue to different PR and did that as temp. solution. :(

File: packages/main/src/services/haveno.ts (hunk 3 above)
Comment (abouolia) [reply packages/main/src/services/haveno.ts]: NO 😄

File: packages/main/src/services/haveno.ts (hunk 1 above)
Comment (schowdhuri) [reply packages/main/src/services/haveno.ts:20]: Gotcha. Let's roll with it for now

File: package.json [hunk 11]
```diff
@@ -96,6 +97,7 @@
     "joi": "^17.6.0",
     "jsonwebtoken": "^8.5.1",
     "lodash": "^4.17.21",
+    "qrcode": "^1.5.0",
```
Comment (schowdhuri) [inline package.json:100]: Ina future iteration I'd like to remove one of the QR libraries. Doesn't make sense to be using two

**Review by schowdhuri — APPROVED** [review]
Let's ship it! 🚀

**Pull request merged**